routes/dms.js

```diff
@@ -148,8 +156,16 @@ module.exports = function () {
       } else if (resource.format === 'pdf') {
         view.specType = 'document'
       }
+      
+      // Determine when to show chart builder
+      const chartBuilderFormats = ['csv', 'tsv']
+      
+      if (chartBuilderFormats.includes(resource.format)) controls.showChartBuilder = true
 
-      datapackage.views.push(view)
+      // displayItem per resource
+      const displayItem = JSON.stringify({resources: [resource], views: [view], controls}).replace(/'/g, "&#x27;")
```

Comment:
@starsinmypockets `resources` item should be a property of a `view`

Reply:
This would make sense to me:

```
view. showChartBuilder = false/true
datapackage.views.push(view)
```

* `resources` property is already defined in the `view` object so you don't need it here
* `showChartBuilder` should be a property of `view` object since it describes a view

Reply:
@starsinmypockets I've just checked PR in Data-Explorer and looks like you did it right there:

```
{
  resources: [...].
  views: [
    {
      resources: [...],
      controls: {
        showChartBuilder: true/false
      },
      specType: ...,
    }
  ]
}
```

Reply:
I think what we want here is a single resource and a single view (for now) per resource, so not passing the entire datapackage, but building a smaller object to instantiate the explorer.

Reply:
And in data-explorer:
```
data-datapackage='{"name":"test","resources":[{"name":"datastore-test","id":"f40147ba-8e83-438e-9e89-bfddd0cba81f","format":"csv","datastore_active":true,"path":"https://www.energidataservice.dk/datastore/dump/f40147ba-8e83-438e-9e89-bfddd0cba81f"}],"views":[{"specType":"table","resources":["datastore-test"]}], "controls": {"showChartBuilder": true}}'
```

Reply:
@anuveyatsu The idea being that controls are a part of the data-explorer config, and we load a single data resource and view object per explorer instance.

Reply:
@starsinmypockets ok, that makes sense. However, I think in this case, we could introduce a new property called `dataexplorers` and use it instead of `views`. What do you think? I guess it shouldn't be a big change.

Also, `controls` should still be part of the view object as per implementation in data-explorer I think.

Reply:
@anuveyatsu I'm not too concerned about the nomenclature - I think `dataExplorers` would be good. I think displayItem is good because we could also add an attribute like `displayItem.renderer = 'dataExplorer'` (loads data-explorer) or `displayItem.renderer = 'simple'` just loads datapackage-views for a single view, which is something Rufus wants and I tend to agree (no 100% but 80% :) ).


--- philosophy:
I think we need to have a longer conversation about improving the interface between these components. Probably they are small changes but I think we need to agree on better nomenclature: "views" is very ambiguous, and difficult to discuss in different contexts. 

Also I think that we are sort of abusing the datapackage.json concept. I think that the datapackage.json is a great idea and should represent an immutable manifest attached to a dataset. I don't think it be mutated to fit the needs of our packages. The packages should have a clear interface and the config objects for each package (datapackage-js, querybuilder, data-explorer, etc) while being possibly based on datapackage.json should not _be_ the datapackage.json. I think that we should do some more thinking on this and come up with a plan of action -- it's obviously risky to change the interface but I think we could address some technical debt here and make further development easier.

Here is the hackmd from the discussion with rufus about this stuff yesterday
https://hackmd.io/BGFPNZNfT1CDIMYecXMcoQ?both#Data-Display-algorithm-%E2%80%A6

Reply:
@starsinmypockets the problem with merging this into master is that it will break other projects since you're making a significant change into views property. I think `views` should follow spec but we can add optional properties for our needs. In this case we are changing entire structure so I think it isn't `views` anymore but `dataexplorers`